tasty/src/dotty/tools/tasty/TastyFormat.scala

```diff
@@ -263,8 +264,25 @@ Standard Section: "Comments" Comment*
 object TastyFormat {
 
   final val header: Array[Int] = Array(0x5C, 0xA1, 0xAB, 0x1F)
-  val MajorVersion: Int = 27
-  val MinorVersion: Int = 0
+
+  /** Each increment of the MajorVersion begins a new series of backward compatible TASTy versions
+   *  - The MajorVersion is and will always be the first value in a TASTy document after the header bytes.
+   *  - A TASTy document can then be further parsed if and only if the MajorVersion read
+   *    from a TASTy file is equal to or less than this value.
+   *  - i.e. `MajorVersion` and `header` are the only stable parts of TASTy that can be used to decide how to parse
+   *    the rest of the document.
```

Comment:
I think we have to give a much stronger guarantee for the header. IMO we should guarantee that the header (the 3 version numbers and the version string) will never change (at least, never after RC1). Otherwise, we can't report good error messages if the major version is not the same.

Reply:
That's a good point, originally this comment was written when parsing `29` as the major number would immediately throw, but now I first parse the major, minor, experimental and compiler version string. I will update the comment to reflect the implementation.